src/bors/handlers/review.rs

```diff
@@ -77,6 +79,26 @@ pub(super) async fn handle_pull_request_edited(
     notify_of_edited_pr(&repo_state, pr_number, &payload.pull_request.base.name).await
 }
 
+pub(super) async fn handle_push_to_pull_request(
+    repo_state: Arc<RepositoryState>,
+    db: Arc<PgDbClient>,
+    payload: PullRequestPushed,
+) -> anyhow::Result<()> {
+    let pr = &payload.pull_request;
+    let pr_model = db
+        .get_or_create_pull_request(repo_state.repository(), pr.number)
+        .await?;
+
+    if !pr_model.is_approved() {
+        return Ok(());
+    }
+
+    let pr_number = pr_model.number;
+    db.unapprove(repo_state.repository(), pr_number).await?;
+    handle_label_trigger(&repo_state, pr_number, LabelTrigger::Unapproved).await?;
```

Comment:
This seems exactly the same as in the case when the base changes. Maybe we can introduce a helper function for unapproving a PR if it's approved?

Reply:
I don't really like how this turned out https://github.com/rust-lang/bors/pull/120/commits/7f9606d8af1f89fe1043f49b4e1906378d4b81af. What do you think?